addons/medical_gui/functions/fnc_onKeyDown.sqf

```diff
@@ -40,39 +45,44 @@ private _allCategories = ["triage", "examine"] + _visibleCategories + ["toggle"]
 private _keyCategoryPairs = _allCategories createHashMapFromArray NUMBER_KEYS;
 
 private _temp_category = "";
+private _temp_idc = 0;
 
 switch (true) do {
 // Dynamically assign number keys to visible categories
     {
         _temp_category = _x; // _x does not exist inside case code
-        case (_keyPressed == _y): {
-            GVAR(selectedCategory) = _temp_category;
-            if (_temp_category == "toggle") then {
-                call FUNC(handleToggle);
-            }
+        _temp_idc = IDC_TRIAGE + (ALL_CATEGORIES find _temp_category) * 10;
+        case (_keyPressed == _y && {GVAR(selectedCategory) != _temp_category}): {
+            if (ctrlEnabled _temp_idc) then {
+                if (_temp_category == "toggle") then {
+                    call FUNC(handleToggle);
+                } else {
+                    GVAR(selectedCategory) = _temp_category;
+                };
+            };
```

Comment:
If you hit a button for a category that is not enabled then it won't play a sound
e.g. Medication when head selected
```suggestion
            } else {
                _return = false;
            };
```